app/views/media/_buttons.html.erb

```diff
@@ -0,0 +1,18 @@
+<span class="buttons">
+	<ul class="row small-up-1 medium-up-2">
+		<li class="column">
+			<%= link_to(image_tag("/assets/upvote_cupid_dart_heart.png"), image_path(upvote_path(movie.id)), action: :upvote, method: :post) %>
```

Comment:
The `action: :upvote` part here is unnecessary.

Usually we only specify the `action` when we're using a hash to indicate what URL a link/button/form will go to (e.g. `{controller: 'media', action: 'show', id: 5}`). Here we are already using `image_path(...)` to indicate the URL and the `action: :upvote` part does not impact that at all -- it actually just becomes an HTML attribute on the `<a>` tag generated by `link_to`.